builtin/providers/aws/resource_aws_cloudwatch_logs_subscription_filter.go

```diff
@@ -0,0 +1,341 @@
+package aws
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"math/rand"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/cloudwatchlogs"
+	"github.com/aws/aws-sdk-go/service/kinesis"
+	"github.com/aws/aws-sdk-go/service/lambda"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+// Number of times to retry if a throttling- or test message exception occurs
+const CLOUDWATCH_LOGS_SUBSCRIPTION_FILTER_MAX_THROTTLE_RETRIES = 10
+
+// How long to sleep when a throttle-event happens
+const CLOUDWATCH_LOGS_SUBSCRIPTION_FILTER_THROTTLE_SLEEP_MILLISECONDS = 8000
+
+func resourceAwsCloudwatchLogsSubscriptionFilter() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCloudwatchLogsSubscriptionFilterCreate,
+		Read:   resourceAwsCloudwatchLogsSubscriptionFilterRead,
+		Update: resourceAwsCloudwatchLogsSubscriptionFilterUpdate,
+		Delete: resourceAwsCloudwatchLogsSubscriptionFilterDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"destination": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"filter_pattern": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
```

Comment:
According to docs http://docs.aws.amazon.com/cli/latest/reference/logs/put-subscription-filter.html most of the parameters should be updatable - are you sure we need `ForceNew` here?